IntrinsicGradedModules/gap/FpGradedModulesByFreyd.gi

```diff
@@ -209,16 +209,23 @@ InstallMethod( FpGradedLeftModules,
         
     end );
     
-    ##
-    AddMultiplyWithElementOfCommutativeRingForMorphisms( P,
-      function( r, alpha )
-        local mat;
-        
-        mat := UnderlyingHomalgMatrix( alpha );
+    if HasIsCommutative( graded_ring ) and IsCommutative( graded_ring ) then
         
-        return GradedRowOrColumnMorphism( Source( alpha ), ( r / HomalgRing( mat ) ) * mat, Range( alpha ) );
+        SetIsLinearCategoryOverCommutativeRing( P, true );
+        SetCommutativeRingOfLinearCategory( P, graded_ring );
```

Comment:
This is the wrong ring. You can find the correct ring [here](https://github.com/homalg-project/CategoricalTowers/blob/1aacab2160331a4954ef249e70a8866397f96f7c/IntrinsicGradedModules/gap/FpGradedModulesByFreyd.gi#L40-L47).